qualification/tied_array_channelised_voltage/test_weights.py

```diff
@@ -61,19 +61,27 @@ async def test_weight_mapping(
 
     pdf_report.step("Set eq gains to select one channel per input.")
     async with asyncio.TaskGroup() as tg:
-        for input_label, channel in zip(receiver.input_labels, channels):
+        for input_label, channel in zip(receiver.input_labels, all_channels_under_test):
             gains = [0.0] * receiver.n_chans
             gains[channel] = 1.0
             tg.create_task(client.request("gain", "antenna-channelised-voltage", input_label, *gains))
             pdf_report.detail(f"Set input {input_label} to pass through only channel {channel}.")
 
-    pdf_report.step("Test all inputs.")
-    for input_idx, (input_label, channel) in enumerate(zip(receiver.input_labels, channels)):
+    pdf_report.step("Test a random selection of inputs.")
+    # NOTE: A full sweep of `all_channels_under_test` takes up too much time during the qualification run.
+    # We test a random selection of the full list, and half the full set at most, to speed that process up.
+    n_inputs_to_test = rng.integers(receiver.n_inputs // 4, max(receiver.n_inputs // 2, 1), dtype=int)
```

Comment:
What is the purpose of randomising the *number* of channels to test? Choosing the channels and inputs themselves randomly makes it extremely unlikely that we'll pick a subset that happens to work (e.g. the first n), but I don't see a benefit to varying the strength of the test randomly.

My suggestion would be to just fix something like
```suggestion
    n_inputs_to_test = min(receiver.n_inputs, 16)
```
which would ensure that small configurations get completely tested while bounding the testing time for larger configurations.